recipes/sbp/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+import os
+
+from textwrap import dedent
+
+import yaml
+
+from conans import ConanFile, CMake, tools
+
+
+class SbpConan(ConanFile):
+    name = "sbp"
+    license = "MIT"
+    author = "Swift Navigation <dev@swiftnav.com>"
+    url = "https://github.com/swift-nav/libsbp"
+    description = "Swift Binary Protocol client library"
+    topics = ("gnss",)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": True, "fPIC": True}
+    generators = "cmake"
+    generators = "cmake"
+    exports = "submoduledata.yml"
+    exports_sources = "c"
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder
+        )
+
+        submodule_filename = os.path.join(self.recipe_folder, "submoduledata.yml")
+        with open(submodule_filename, "r") as submodule_stream:
+            submodules_data = yaml.load(submodule_stream)
+            for path, submodule in submodules_data["submodules"][self.version].items():
+                submodule_data = {
+                    "url": submodule["url"],
+                    "sha256": submodule["sha256"],
+                    "destination": os.path.join(self._source_subfolder, path),
+                    "strip_root": True,
+                }
+
+                tools.get(**submodule_data)
+
+        tools.replace_in_file(
+            "{}/c/CMakeLists.txt".format(self._source_subfolder),
+            "project(libsbp C CXX)",
+            dedent(
+                """
+                project(libsbp C CXX)
```

Comment:
Instead of patching, if the project supports it, you can also include the project as a subproject.

Reply:
Do you have an example in `conan-center-index` where that is done?

Reply:
Sure, there are a lot.
e.g. `rtmidi`: it has a `CMakeLists.txt` that included `conanbuildinfo.cmake`, calls `conan_basic_setup` and does `add_subdirectory`
https://github.com/conan-io/conan-center-index/tree/master/recipes/rtmidi/all